sdk/python/tests/README.md

```diff
@@ -0,0 +1,341 @@
+        name="entity_mapping",
+        features=[
+            feature_views.location.with_name("origin").with_join_key_map(
+                {"location_id": "origin_id"}
+            ),
+            feature_views.location.with_name("destination").with_join_key_map(
+                {"location_id": "destination_id"}
+            ),
+        ],
+    )
+
+    store.apply(
+        [
+            driver(),
+            customer(),
+            location(),
+            feature_service,
+            feature_service_entity_mapping,
+            *feature_views.values(),
+        ]
+    )
+    # ... more test code
+
+    job_from_df = store.get_historical_features(
+        entity_df=entity_df_with_request_data,
+        features=[
+            "driver_stats:conv_rate",
+            "driver_stats:avg_daily_trips",
+            "customer_profile:current_balance",
+            "customer_profile:avg_passenger_count",
+            "customer_profile:lifetime_trip_count",
+            "conv_rate_plus_100:conv_rate_plus_100",
+            "conv_rate_plus_100:conv_rate_plus_100_rounded",
+            "conv_rate_plus_100:conv_rate_plus_val_to_add",
+            "order:order_is_success",
+            "global_stats:num_rides",
+            "global_stats:avg_ride_length",
+            "field_mapping:feature_name",
+        ],
+        full_feature_names=full_feature_names,
+    )
+
+    if job_from_df.supports_remote_storage_export():
+        files = job_from_df.to_remote_storage()
+        print(files)
+        assert len(files) > 0  # This test should be way more detailed
+
+    start_time = datetime.utcnow()
+    actual_df_from_df_entities = job_from_df.to_df()
+    # ... more test code
+
+    validate_dataframes(
+        expected_df,
+        table_from_df_entities,
+        keys=[event_timestamp, "order_id", "driver_id", "customer_id"],
+    )
+    # ... more test code
+```
+{% endtab %}
+{% endtabs %}
+
+- The key fixtures are the `environment` and `universal_data_sources` fixtures, which are defined in the `feature_repos` directories and the `conftest.py` file. This by default pulls in a standard dataset with driver and customer entities(that we have pre-defined), certain feature views, and feature values.
+    - The `environment` fixture sets up a feature store, parametrized by the provider and the online/offline store. It allows the test to query against that feature store without needing to worry about the underlying implementation or any setup that may be involved in creating instances of these datastores.
+    - Each fixture creates a different integration test with its own `IntegrationTestRepoConfig` which is used by pytest to generate a unique test testing one of the different environments that require testing.
+
+- Feast tests also use a variety of markers:
+    - The `pytest.mark.integration` marker is used to designate integration tests which will cause the test to be run when you call `make test-python-integration`.
```

Comment:
nit: use @ consistently (this line omits it, next line uses it)

Reply:
fixed